### ISSUE TITLE
Pretrained weights for ResNet[18, 34, 50, 101 ]

### ISSUE THREAD

**Issue by longyuxi** [pytorch/vision#8749]
### 🐛 Describe the bug

Hi,

I have been trying to run the pretrained ResNet models. The model weights seem to be incorrect. Below is a code to reproduce the erroneous results:

```
import torch
from torchvision.models import resnet18, ResNet18_Weights
from PIL import Image

resnet = resnet18(weights=ResNet18_Weights.IMAGENET1K_V1)
preprocess = ResNet18_Weights.IMAGENET1K_V1.transforms()

# !wget "https://github.com/pytorch/hub/raw/master/images/dog.jpg"
input_image = Image.open('dog.jpg')

# !wget https://upload.wikimedia.org/wikipedia/commons/b/b6/Felis_catus-cat_on_snow.jpg -O cat.jpg
# input_image = Image.open('cat.jpg')

input_tensor = preprocess(input_image)
input_batch = input_tensor.unsqueeze(0) # create a mini-batch as expected by the model

with torch.no_grad():
    output = resnet(input_batch)

probabilities = torch.nn.functional.softmax(output[0], dim=0)
# !wget https://raw.githubusercontent.com/pytorch/hub/master/imagenet_classes.txt
with open("imagenet_classes.txt", "r") as f:
    categories = [s.strip() for s in f.readlines()]

# Show top categories per image
top5_prob, top5_catid = torch.topk(probabilities, 5)
for i in range(top5_prob.size(0)):
    print(categories[top5_catid[i]], top5_prob[i].item())
```

Output is the same for both "cat.jpg" and "dog.jpg" for ResNet18:

```
bucket 0.008743884041905403
plunger 0.006772771943360567
hook 0.005883160978555679
paper towel 0.005243286956101656
ashcan 0.005110109690576792
```

These predictions are clearly incorrect. Through a noncomprehensive testing, the garbage output occurs for the model weights:

```
ResNet18_Weights.IMAGENET1K_V1
ResNet34_Weights.IMAGENET1K_V1
ResNet50_Weights.IMAGENET1K_V1
ResNet101_Weights.IMAGENET1K_V1
```

while the output for the following model weights are correct:

```
ResNet50_Weights.IMAGENET1K_V2
ResNet101_Weights.IMAGENET1K_V2
```

My guess is that the pretrained weight files are linked incorrectly for the V1 models.

### Versions

Collecting environment information...
PyTorch version: 2.5.1
Is debug build: False
CUDA used to build PyTorch: 12.4
ROCM used to build PyTorch: N/A

OS: Rocky Linux 9.4 (Blue Onyx) (x86_64)
GCC version: (GCC) 11.4.1 20231218 (Red Hat 11.4.1-3)
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.34

Python version: 3.12.3 | packaged by conda-forge | (main, Apr 15 2024, 18:38:13) [GCC 12.3.0] (64-bit runtime)
Python platform: Linux-5.14.0-427.42.1.el9_4.x86_64-x86_64-with-glibc2.34
Is CUDA available: True
CUDA runtime version: 12.4.131
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: 
GPU 0: NVIDIA A10
GPU 1: NVIDIA A10

Nvidia driver version: 550.54.15
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                         x86_64
CPU op-mode(s):                       32-bit, 64-bit
Address sizes:                        46 bits physical, 57 bits virtual
Byte Order:                           Little Endian
CPU(s):                               96
On-line CPU(s) list:                  0-95
Vendor ID:                            GenuineIntel
Model name:                           Intel(R) Xeon(R) Gold 6342 CPU @ 2.80GHz
CPU family:                           6
Model:                                106
Thread(s) per core:                   2
Core(s) per socket:                   24
Socket(s):                            2
Stepping:                             6
CPU(s) scaling MHz:                   98%
CPU max MHz:                          3500.0000
CPU min MHz:                          800.0000
BogoMIPS:                             5600.00
Flags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 intel_ppin ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect wbnoinvd dtherm ida arat pln pts vnmi avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid fsrm md_clear pconfig flush_l1d arch_capabilities
Virtualization:                       VT-x
L1d cache:                            2.3 MiB (48 instances)
L1i cache:                            1.5 MiB (48 instances)
L2 cache:                             60 MiB (48 instances)
L3 cache:                             72 MiB (2 instances)
NUMA node(s):                         2
NUMA node0 CPU(s):                    0,2,4,6,8,10,12,14,16,18,20,22,24,26,28,30,32,34,36,38,40,42,44,46,48,50,52,54,56,58,60,62,64,66,68,70,72,74,76,78,80,82,84,86,88,90,92,94
NUMA node1 CPU(s):                    1,3,5,7,9,11,13,15,17,19,21,23,25,27,29,31,33,35,37,39,41,43,45,47,49,51,53,55,57,59,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95
Vulnerability Gather data sampling:   Mitigation; Microcode
Vulnerability Itlb multihit:          Not affected
Vulnerability L1tf:                   Not affected
Vulnerability Mds:                    Not affected
Vulnerability Meltdown:               Not affected
Vulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT vulnerable
Vulnerability Reg file data sampling: Not affected
Vulnerability Retbleed:               Not affected
Vulnerability Spec rstack overflow:   Not affected
Vulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop
Vulnerability Srbds:                  Not affected
Vulnerability Tsx async abort:        Not affected

Versions of relevant libraries:
[pip3] numpy==2.1.3
[pip3] torch==2.5.1
[pip3] torchaudio==2.5.1
[pip3] torchvision==0.20.1
[pip3] triton==3.1.0
[conda] blas                      2.116                       mkl    conda-forge
[conda] blas-devel                3.9.0            16_linux64_mkl    conda-forge
[conda] cuda-cudart               12.4.127                      0    nvidia
[conda] cuda-cupti                12.4.127                      0    nvidia
[conda] cuda-libraries            12.4.1                        0    nvidia
[conda] cuda-nvrtc                12.4.127                      0    nvidia
[conda] cuda-nvtx                 12.4.127                      0    nvidia
[conda] cuda-opencl               12.6.77                       0    nvidia
[conda] cuda-runtime              12.4.1                        0    nvidia
[conda] libblas                   3.9.0            16_linux64_mkl    conda-forge
[conda] libcblas                  3.9.0            16_linux64_mkl    conda-forge
[conda] libcublas                 12.4.5.8                      0    nvidia
[conda] libcufft                  11.2.1.3                      0    nvidia
[conda] libcurand                 10.3.7.77                     0    nvidia
[conda] libcusolver               11.6.1.9                      0    nvidia
[conda] libcusparse               12.3.1.170                    0    nvidia
[conda] liblapack                 3.9.0            16_linux64_mkl    conda-forge
[conda] liblapacke                3.9.0            16_linux64_mkl    conda-forge
[conda] libnvjitlink              12.4.127                      0    nvidia
[conda] mkl                       2022.1.0           h84fe81f_915    conda-forge
[conda] mkl-devel                 2022.1.0           ha770c72_916    conda-forge
[conda] mkl-include               2022.1.0           h84fe81f_915    conda-forge
[conda] numpy                     2.1.3           py312h58c1407_0    conda-forge
[conda] pytorch                   2.5.1           py3.12_cuda12.4_cudnn9.1.0_0    pytorch
[conda] pytorch-cuda              12.4                 hc786d27_7    pytorch
[conda] pytorch-mutex             1.0                        cuda    pytorch
[conda] torchaudio                2.5.1               py312_cu124    pytorch
[conda] torchtriton               3.1.0                     py312    pytorch
[conda] torchvision               0.20.1              py312_cu124    pytorch

**Comment by longyuxi** [comment]
For convenience, here's the code showing that the weights for `ResNet50_Weights.IMAGENET1K_V2` are correct:

```
import torch
from torchvision.models import resnet50, ResNet50_Weights
from PIL import Image

resnet = resnet50(weights=ResNet50_Weights.IMAGENET1K_V2)
preprocess = ResNet50_Weights.IMAGENET1K_V2.transforms()

# !wget "https://github.com/pytorch/hub/raw/master/images/dog.jpg"
input_image = Image.open('dog.jpg')

# !wget https://upload.wikimedia.org/wikipedia/commons/b/b6/Felis_catus-cat_on_snow.jpg -O cat.jpg
# input_image = Image.open('cat.jpg')

input_tensor = preprocess(input_image)
input_batch = input_tensor.unsqueeze(0) # create a mini-batch as expected by the model

with torch.no_grad():
    output = resnet(input_batch)

probabilities = torch.nn.functional.softmax(output[0], dim=0)
# !wget https://raw.githubusercontent.com/pytorch/hub/master/imagenet_classes.txt
with open("imagenet_classes.txt", "r") as f:
    categories = [s.strip() for s in f.readlines()]

# Show top categories per image
top5_prob, top5_catid = torch.topk(probabilities, 5)
for i in range(top5_prob.size(0)):
    print(categories[top5_catid[i]], top5_prob[i].item())
```

Output:

```
Eskimo dog 0.003977478947490454
Samoyed 0.003522714599967003
Siberian husky 0.0031752213835716248
West Highland white terrier 0.00291797099635005
white wolf 0.0022370663937181234
```

**Comment by NicolasHug** [comment]
Thanks for the report @longyuxi .

That gave me a little heart attack, but the problem is that `eval()` wasn't called in your snippet above. Calling no_grad() is not enough, partially because of the BatchNorm layers.

With `resnet.eval()` called on the resnet50 v1 weights:

```
Samoyed 0.8732956051826477
Pomeranian 0.030270902439951897
white wolf 0.019671238958835602
keeshond 0.011073538102209568
Eskimo dog 0.00920429639518261
```

**Comment by longyuxi** [comment]
@NicolasHug Ah, thank you very much! Sorry for freaking you out and I wish you a nice rest of the day.
